torch/__init__.py

```diff
@@ -247,6 +230,7 @@ def manager_path():
 for name in dir(_C._VariableFunctions):
     globals()[name] = getattr(_C._VariableFunctions, name)
 
+set_default_tensor_type(globals()['float32'])
```

Comment:
I found this much easier to follow when all the type creation + default type logic was in C++; can this not still be done in C++? (I know the implementation would be different because you got rid of the extra PyTypeObject).

Reply:
I think we might be able to move it back to C++. I'll check.